### PR TITLE
Deprecate JAVA_HOME preferring LS_JAVA_HOME

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Deprecate the usage of `JAVA_HOME` preferring `LS_JAVA_HOME`

## What does this PR do?

Simply print deprecation notices on stdout when Logstash is starting with `JAVA_HOME` instead of `LS_JAVA_HOME`

## Why is it important/What is the impact to the user?

Helps the user to migrate from `JAVA_HOME` to `JAVA_HOME`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] ~~I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: bin/logstash.lib.sh [hunk 1]
```diff
@@ -111,6 +111,7 @@ setup_java() {
       else
         echo "Invalid JAVA_HOME, doesn't contain bin/java executable."
       fi
+      echo "DEPRECATION: Using JAVA_HOME, please configure LS_JAVA_HOME instead."
```
Comment (roaksoax) [inline bin/logstash.lib.sh:114]: Now that I read this, I think we need to improve it.

DEPRECATION: The use of JAVA_HOME is now deprecated and will be soon removed, please configure LS_JAVA_HOME instead.

Now, we could also say in which release we would deprecate, e.g.:

DEPRECATION: The use of JAVA_HOME is now deprecated and will be removed starting from 8.0. Please configure LS_JAVA_HOME instead

File: bin/setup.bat [hunk 2]
```diff
@@ -32,6 +32,7 @@ if defined LS_JAVA_HOME (
   if exist "%LS_HOME%\jdk" (
     echo WARNING: Using JAVA_HOME while Logstash distribution comes with a bundled JDK.
   )
+  echo DEPRECATION: Using JAVA_HOME, please configure LS_JAVA_HOME instead.
```
Comment (roaksoax) [inline bin/setup.bat:35]: Same here.

File: bin/setup.bat [hunk 3]
```diff
@@ -32,6 +32,7 @@ if defined LS_JAVA_HOME (
   if exist "%LS_HOME%\jdk" (
     echo WARNING: Using JAVA_HOME while Logstash distribution comes with a bundled JDK.
```
Comment (roaksoax) [inline bin/setup.bat:33]: Shouldn't this also be updated? and/or also take LS_JAVA_HOME into consideration?

File: docs/static/jvm.asciidoc [hunk 4]
```diff
@@ -53,8 +53,7 @@ Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.1+13-LTS, mixed mode)
 variable must be set for {ls} to operate correctly.
 
 If {ls} doesn't find `LS_JAVA_HOME` it tries to fall back to `JAVA_HOME`.
-Note that this fallback behavior is experimental and the usage of `JAVA_HOME`
-could be subject to removal in later releases.
+The usage of `JAVA_HOME` is considered deprecated.
```
Comment (roaksoax) [inline docs/static/jvm.asciidoc:56]: The usage of `JAVA_HOME` is now considered deprecated in favor of `LS_JAVA_HOME`.

**Review by roaksoax — CHANGES_REQUESTED** [review]

File: bin/logstash.lib.sh (hunk 1 above)
Comment (andsel) [reply bin/logstash.lib.sh:114]: Sounds better

File: bin/setup.bat (hunk 3 above)
Comment (andsel) [reply bin/setup.bat:33]: The deprecation notice is written at subsequent message, so I wouldn't write it twice. This warning is specific to for the usage of env variable when the LS package comes with a bundled JDK.

With this message the user sees on his console:
```
WARNING: Using JAVA_HOME while Logstash distribution comes with a bundled JDK.
DEPRECATION: The use of JAVA_HOME is now deprecated and will be removed starting from 8.0. Please configure LS_JAVA_HOME instead.
```
So he's fully informed of the deprecation and the usage of a system JDK while Logstash is bundled with a JDK.

**Review by roaksoax — APPROVED** [review]

**Pull request merged**